server/src/main/java/io/crate/execution/dml/TranslogIndexer.java

```diff
@@ -113,9 +139,13 @@ private IndexDocumentBuilder populateLuceneFields(BytesReference source) throws
             Object castValue = valueForInsert(indexer.dataType, entry.getValue());
             if (castValue != null) {
                 indexer.valueIndexer.indexValue(castValue, docBuilder);
-                if (tableIndexSources.containsKey(column)) {
-                    addIndexField(docBuilder, tableIndexSources.get(column), castValue);
+                for (Indexer.IndexColumn indexColumn: indexColumns) {
+                    for (Input<?> input: indexColumn.inputs()) {
+                        TranslogInput translogInput = (TranslogInput) input;
+                        translogInput.prepareValue(castValue, indexer.dataType);
+                    }
                 }
+                addIndexColumns(indexColumns, docBuilder);
             }
```

Comment:
This would add **all** index columns for each table column.
```suggestion
            }
            addIndexColumns(indexColumns, docBuilder);
```

Reply:
Replied in https://github.com/crate/crate/pull/17369#discussion_r1949125424